### DISCUSSION TITLE
Invoke Lambda without public url

### DISCUSSION THREAD

**Discussion by xandernewton** [awslabs/aws-lambda-web-adapter#535] [Q&A]
Is there a way to invoke the lambda through boto3 and not by calling the public URL of lambda? Can the public URL of the lambda be turned off?

**Comment by sladg** [comment]
SDK has `invoke` function so one lambda can invoke another lambda for example. this happens directly and does not require function URL. However, wouldn't that defeat the purpose of this package? if you have control over the invocation, you have a control over the payload as well I assume?

**Comment by bnusunny** [comment]
@xandernewton Are you referring to the Lambda Function URL or the Lambda service endpoint? 

For Lambda Function URL, you can simply disable it. For Lambda service endpoint, you can create private link in your vpc to access it. But the Lambda Service Endpoint can't be turned off.

**Comment by xandernewton** [comment]
I might have misunderstood how the library works. What I want to able to do is to define the lambda function in CDK using the lambda docker construct and then once the lambda is setup, just invoking as normal with boto3 with a json payload. In the fast API examples if you need to call an endpoint with a url. I'm wondering if this can be achieved with a json payload. From what I gather, currently once the lambda is deployed it has a public url?

**Reply by bnusunny** [reply]
You can turn off the Function URL or add IAM auth to make it not open to everyone. 

```yaml
      FunctionUrlConfig:
            AuthType: AWS_IAM
            InvokeMode: RESPONSE_STREAM
```
